src/tools/split.cpp

```diff
@@ -0,0 +1,103 @@
+#include "split.h"
+#include "editorview.h"
+#include "meshitem.h"
+#include "gripitem.h"
+#include <QMouseEvent>
+#include <QDebug>
+#include <limits>
+
+SplitTool::SplitTool()
+: BaseTool()
+{
+}
+
+void SplitTool::activated(EditorView* editor)
+{
+  if (marker == nullptr) {
+    marker = new MarkerItem();
+    editor->scene()->addItem(marker);
+    marker->setHighlight(QColor("#FF4444"));
+  }
+  GripItem* closestGrip = editor->snapGrip();
+  EdgeItem* closestEdge = editor->snapEdge();
+
+  if (closestEdge || closestGrip) {
+    marker->setVisible(true);
+    if (closestGrip) {
+      marker->setBrush(closestGrip->color());
+    }
+    else {
+      marker->setBrush(closestEdge->colorAt(editor->snapPosition()));
+    }
+    marker->setPos(editor->snapPosition());
+  }
+  else {
+    marker->setVisible(false);
+  }
+}
+
+void SplitTool::deactivated(EditorView* editor)
+{
+    marker->setVisible(false);
+}
+
+bool SplitTool::mousePressEvent(EditorView* editor, QMouseEvent* event)
+{
+  MeshItem* mesh = editor->activeMesh();
+  GripItem* oldActive = editor->activeVertex();
+  GripItem* closestGrip = editor->snapGrip();
+  EdgeItem* closestEdge = editor->snapEdge();
+  if (closestEdge) {
+    if (!closestGrip) {
+      closestEdge->split(editor->snapPosition());
+    }
+    else {
+      editor->setActiveVertex(closestGrip);
+    }
+    if (oldActive) {
+      mesh->splitPolygon(oldActive, editor->activeVertex());
+    }
+  }
+  /* if (grip) { */
```

Comment:
Wow, this is really how the comment-out feature decided to do it? Not `/*` at the beginning and `*/` at the end, not `//` at the beginning of each line...

You can probably just delete it.

Reply:
Whoops! I intended to delete that. Left it there for reference to function names when writing the new implementation.